### ISSUE TITLE
Unexpected existence error

### ISSUE THREAD

**Issue by triska** [aarroyoc/scryer-playground#20]
With the following program:

<pre>
:- use_module(library(lambda)).

p.
</pre>

I unexpectedly get:

<pre>
<b>?- X +\ p.</b>

   error(existence_error(procedure,p/0),p/0).
</pre>

Expected:

<pre>
<b>?- X +\ p.</b>
   true.
</pre>

**Comment by bakaq** [comment]
~This also happens in the Scryer CLI, this is not specific to the playground. I don't understand why `true` is expected here. As I far as I know `(+\)/2` calls the goal that is it's second argument with the free bindings of the first argument. `p/0` isn't defined here, so an `existence_error` seems to be expected.~

**Comment by triska** [comment]
With Scryer, I get:

<pre>
<b>$ scryer-prolog  -f
?- [user].</b>
:- use_module(library(lambda)).

p.

<b>?-  X +\ p.</b>
   true.
</pre>

**Comment by bakaq** [comment]
This is true, I'm being careless again. I think this is maybe a problem with metapredicates, because this works:

```
?- X +\ user:p.
   % The pseudo-toplevel doesn't treat this case correctly for now,
   % but it means the same as "true".
   X = A. 
```

**Comment by bakaq** [comment]
I think that with https://github.com/mthom/scryer-prolog/issues/2255 now solved, updating the WASM artifact to the latest master will solve this.

**Comment by aarroyoc** [comment]
I've updated the version of the playground with the new 0.9.4 release and it keeps happening

**Comment by dcnorris** [comment]
This is really hurting me now, since I have become heavily dependent on
```prolog
?- N +\ (multi_step(X), calculation(X,Y),
         with_huge_unprintable_intermediates(X,Y,Z,P,D,Q),
         length(Q, N)).
```
Seems like a regression since my last Scryer rebuild ~1 month ago?

---

(Sorry if out of scope; I just clicked through blindly without realizing this was a derived project.  In any case, the abovementioned error is suddenly happening in main Scryer build on Linux.)

**Comment by aarroyoc** [comment]
@dcnorris I was confused at first because I did not update the Playground build since the 0.9.4 release

**Comment by triska** [comment]
@aarroyoc, @dcnorris: Could you please try it with the most recent commit https://github.com/mthom/scryer-prolog/commit/dba9c9a0a79f5199ad1770edb57fd2531ee270ee? Thank you a lot!

**Comment by dcnorris** [comment]
This works now, so nice to have it back!
```prolog
?- N+\(setof(X, (X in 0..9, indomain(X)), Xs), length(Xs, N)).
%@    N = 10.
```
